server/db/index.js

```diff
@@ -1,8 +1,16 @@
-const mongoose = require('mongoose')
+const mongoose = require('mongoose');
 
-const db = mongoose.connect(process.env.MONGO_URI_DEV, {
-  useNewUrlParser: true,
-  useUnifiedTopology: true
-}, () => console.log("mongoose connected"))
+exports.connect = (URI, cb) => new Promise((resolve, reject) => {
```

Comment:
What is the purpose of callback parameter here? seems like you've already got it structured as a promise and are error handling within function body with a `try...catch` block

Reply:
it was there in case user wanted to access the connection data, but it makes more sense to expose it with the resolve. Will make change to handle both